cmd/gardenlet/app/migration.go

```diff
@@ -106,3 +115,35 @@ func migrateDeprecatedTopologyLabels(ctx context.Context, log logr.Logger, seedC
 
 	return flow.Parallel(taskFns...)(ctx)
 }
+
+// deleteStaleAlertmanagerVPAs deletes VPAs with name "alertmanager-vpa" in the Seed cluster.
+// In https://github.com/gardener/gardener/pull/9257/files#diff-31f2d707167e32b68a064c12fe955a5f2e7d6668e4de445ea9bf6b4e125e6091R95-R103 we forgot to delete
+// the VPA related to the old alertmanager deployment (https://github.com/gardener/gardener/pull/9257/files#diff-48abc7fddac745815a412837ac95081265c7ffcd80d5cbf3b2ec1454b2a4068aL160-L175).
+//
+// TODO(ialidzhikov): Remove this function after v1.106 has been released.
+func deleteStaleAlertmanagerVPAs(ctx context.Context, log logr.Logger, seedClient client.Client) error {
+	vpas := &vpaautoscalingv1.VerticalPodAutoscalerList{}
+	if err := seedClient.List(ctx, vpas); err != nil {
+		if meta.IsNoMatchError(err) {
+			log.Info("Received a 'no match error' while trying to list VPAs. Will assume that the VPA CRD is not yet installed (for example new Seed creation) and will skip cleaning up stale alertmanager VPAs")
+			return nil
+		}
+
+		return err
+	}
+
+	return utilclient.ApplyToObjects(ctx, vpas, func(ctx context.Context, obj client.Object) error {
+		vpa, ok := obj.(*vpaautoscalingv1.VerticalPodAutoscaler)
+		if !ok {
+			return fmt.Errorf("expected *vpaautoscalingv1.VerticalPodAutoscaler but got %T", obj)
+		}
+
+		if vpa.Name == "alertmanager-vpa" {
+			if err := kubernetesutils.DeleteObject(ctx, seedClient, vpa); err != nil {
+				return fmt.Errorf("failed to delete VPA %s: %w", client.ObjectKeyFromObject(vpa), err)
+			}
+		}
+
+		return nil
+	})
```

Comment:
```suggestion
	return utilclient.ApplyToObjects(ctx, vpas, func(ctx context.Context, obj client.Object) error {
		if obj.GetName() == "alertmanager-vpa" {
			if err := kubernetesutils.DeleteObject(ctx, seedClient, obj); err != nil {
				return fmt.Errorf("failed to delete VPA %s: %w", client.ObjectKeyFromObject(obj), err)
			}
		}

		return nil
	})
```
Do we need the conversion? Can we just use `obj`?

Reply:
Good point, I don't need the conversion to the VPA object. Copy-paste from similar old PR...
I now incorporated your suggestion. Thanks!